app/routes/forum/categories/category/threads/index.js

```diff
@@ -38,7 +39,15 @@ export default class ThreadIndexRoute extends AuthenticatedRoute {
     return this.can.can('show forum/threads');
   }
 
-  model() {
-    return this.modelFor('forum.categories.category');
+  async model(params) {
+    const category = this.modelFor('forum.categories.category');
+    assign(params, {
+      filter: { category: category.id }
+    });
+    const threadsPromise = await this.store.queryPaged('forum/thread', params);
+    return {
+      category,
+      threads: threadsPromise
```

Comment:
Where is this data used? I only see a change for the model, but not a change in how the model works.

Reply:
I believe this is necessary because queryPaged is passed `params`, which if I recall correctly is modified by (or based on) the FilterableAndSortableMixin. If you look at the controller for this route, https://github.com/csvalpha/amber-ui/blob/staging/app/controllers/forum/categories/category/threads/index.js you can see that the queryParams are set to ['search', 'sort'] these are defined in the FilterableAndSortableMixin, and the way queryParams work is defined here: https://guides.emberjs.com/release/routing/query-params/
In the template that belongs to the this route, the search and sort parameters are adjustable for the user through the FilterBar component.
Without passing this code, the `params` are never used in the query to the backend, and the threads that belong to the category are neither sorted nor filtered.